python/pyspark/pandas/frame.py

```diff
@@ -455,10 +456,14 @@ def __init__(  # type: ignore[no-untyped-def]
                 from pyspark.pandas.indexes.base import Index
 
                 if isinstance(index, Index):
-                    raise TypeError(
-                        "The given index cannot be a pandas-on-Spark index. "
-                        "Try pandas index or array-like."
-                    )
+                    if get_option("compute.ops_on_diff_frames"):
+                        index = index.to_pandas()
```

Comment:
I think we should probably create pandas-on-Spark DataFrame with joining (e.g., df1 + df2 when `compute.ops_on_diff_frames` is on) instead of collecting it to the driver side.

Reply:
oh, I misunderstood the task.

in this case, the data should be converted to `SparkDataFrame` or `InternalFrame ` and then joined with the ps.Index, right?

Reply:
I am still a lit confused:

do you mean: